driver-core/src/main/com/mongodb/internal/connection/DefaultConnectionPool.java

```diff
@@ -945,26 +945,13 @@ void tryHandOverOrRelease(final UsageTrackingInternalConnection openConnection)
             }
         }
 
-        /**
-         * @param timeout If {@linkplain Timeout#isInfinite() infinite},
-         *                then the lock is {@linkplain Lock#lockInterruptibly() acquired interruptibly}.
-         */
-        private void tryLock(final Timeout timeout) throws MongoTimeoutException {
-            boolean success;
+        private void lockInterruptibly(final Lock lock) throws MongoTimeoutException {
```

Comment:
Why the change to take the lock as the parameter?  Is there more than lock that we could be using in this context?
Taking the lock as a param it seems more like a utility method and therefore should be static. 

What do you think?

Reply:
> Why the change to take the lock as the parameter? Is there more than lock that we could be using in this context?

The change is to remove the weird asymmetry. Previously the code looked like
```java
lockInterruptibly(); // use a method that uses the `lock` field internally
try {
  ...
} finally {
  lock.unlock(); // use the `lock` field explicitly
}
```
now it looks like
```java
lockInterruptibly(lock); // use the `lock` field explicitly
try {
  ...
} finally {
  lock.unlock(); // use the `lock` field explicitly
}
```

> a utility method and therefore should be static

I 100% agree, and I wish we could write static methods in inner classes, but Java syntax forbids that (this restriction seems to be artificial and not helpful).